src/components/SettingsPage/DisablePasswordButton.js

```diff
@@ -0,0 +1,133 @@
+// @flow
+
+import React, { Fragment, PureComponent } from 'react'
+import { connect } from 'react-redux'
+import { translate } from 'react-i18next'
+import type { T } from 'types/common'
+import { createStructuredSelector } from 'reselect'
+import bcrypt from 'bcryptjs'
+import { setEncryptionKey } from 'helpers/db'
+import { cleanAccountsCache } from 'actions/accounts'
+import { saveSettings } from 'actions/settings'
+import { storeSelector } from 'reducers/settings'
+import type { SettingsState } from 'reducers/settings'
+import { unlock } from 'reducers/application' // FIXME should be in actions
+import Track from 'analytics/Track'
+import CheckBox from 'components/base/CheckBox'
+import Box from 'components/base/Box'
+import Button from 'components/base/Button'
+import PasswordModal from './PasswordModal'
+import DisablePasswordModal from './DisablePasswordModal'
+
+const mapStateToProps = createStructuredSelector({
+  // FIXME in future we should use dedicated password selector and a savePassword action (you don't know the shape of settings)
+  settings: storeSelector,
+})
+
+const mapDispatchToProps = {
+  unlock,
+  cleanAccountsCache,
+  saveSettings,
+}
+
+type Props = {
+  t: T,
+  unlock: () => void,
+  settings: SettingsState,
+  saveSettings: Function,
```

Comment:
Rare case of seeing @gre using `Function` type

Reply:
actually all the code of this PR : I have not written a single line. i just dup the section files into a lot of pieces but preserve almost the same. simplified what i could find.

and this part is actually not refactored. it shouldn't take `settings` and `saveSettings` into parameter. this is heresy ! who knows what `settings` structure is ;)